plugins/catalog-backend/src/catalog/DatabaseItemsCatalog.ts

```diff
@@ -14,17 +14,19 @@
  * limitations under the License.
  */
 
-import { Location } from '../catalog/types';
-import { ComponentDescriptor } from '../descriptors';
-import { readFileLocation } from './fileLocation';
+import { Database } from '../database';
+import { Component, ItemsCatalog } from './types';
 
-export async function readLocation(
-  location: Location,
-): Promise<ComponentDescriptor[]> {
-  switch (location.type) {
-    case 'file':
-      return await readFileLocation(location.target);
-    default:
-      throw new Error(`Unknown type "${location.type}"`);
+export class DatabaseItemsCatalog implements ItemsCatalog {
+  constructor(private readonly database: Database) {}
+
+  async components(): Promise<Component[]> {
+    const items = await this.database.components();
+    return items;
```

Comment:
note that the return type from the database is technically not the same as the one returned by this method - they happen to be the same at this moment, but they will likely diverge and evolve separately, so there will be a translation here from one format to the other